### PR TITLE
K8SPSMDB-244: Add missed unlock in reconcile loop

### PR DESCRIPTION
**Pull request by blez**
[K8SPSMDB-244](https://jira.percona.com/browse/K8SPSMDB-244)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pkg/controller/perconaservermongodb/psmdb_controller.go [hunk 1]
```diff
@@ -150,6 +151,15 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 		RequeueAfter: r.reconcileIn,
 	}
 
+	// PerconaServerMongoDB object is also accessed and changed by a version service's corn job (that run concurrently)
```
Comment (tplavcic) [inline pkg/controller/perconaservermongodb/psmdb_controller.go:154]: `corn job` -> `cron job`
`that run` -> `that runs`

Comment (blez) [reply pkg/controller/perconaservermongodb/psmdb_controller.go]: https://github.com/percona/percona-server-mongodb-operator/pull/393/commits/8b8c5a92a3dcfedeb74621c8f7493e332ed9bf56

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| one-pod | passed |
| storage | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| security-context | passed |
| arbiter | passed |
| operator-self-healing | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |

**Comment by JNKPercona** [comment]
PSMDB operator docker - perconalab/percona-server-mongodb-operator:PR-393-45226d31

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade | passed |
| monitoring | passed |
| self-healing | passed |
| limits | passed |
| upgrade-consistency | passed |
| scaling | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| security-context | passed |
| operator-self-healing | passed |
| arbiter | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| liveness | passed |

**Pull request merged**